Plugins/AWSLambdaDeployer/Plugin.swift

```diff
@@ -0,0 +1,348 @@
+// ===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftAWSLambdaRuntime open source project
+//
+// Copyright (c) 2022 Apple Inc. and the SwiftAWSLambdaRuntime project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftAWSLambdaRuntime project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// ===----------------------------------------------------------------------===//
+
+import Dispatch
+import Foundation
+import PackagePlugin
+
+@main
+struct AWSLambdaPackager: CommandPlugin {
+    func performCommand(context: PackagePlugin.PluginContext, arguments: [String]) async throws {
+
+        let configuration = try Configuration(context: context, arguments: arguments)
+        guard !configuration.products.isEmpty else {
+            throw Errors.unknownProduct("no appropriate products found to deploy")
+        }
+
+        let samExecutable = try context.tool(named: "sam")
+        let samExecutableUrl = URL(fileURLWithPath: samExecutable.path.string)
+        let deploymentDescriptorExecutableUrl = configuration.deployExecutable
+        if configuration.verboseLogging {
+            print("-------------------------------------------------------------------------")
+            print("executables")
+            print("-------------------------------------------------------------------------")
+            print("SAM Executable : \(samExecutableUrl)")
+            print("Deployment Descriptor Executable : \(deploymentDescriptorExecutableUrl)")
+        }
+
+        let currentDirectory = FileManager.default.currentDirectoryPath
+        let samDeploymentDescriptorUrl = URL(fileURLWithPath: currentDirectory)
```

Comment:
not sure we need a URL here. we can probably use Path from the PackagePlugin API

Reply:
I removed all unnecessary usage of URL.  I still use a few when I use `FileManager` or `String` functions that expect an `URL` argument